solr/solrj/src/java/org/apache/solr/common/cloud/SolrZkClient.java

```diff
@@ -256,18 +265,43 @@ protected ZkACLProvider createZkACLProvider() {
     if (!StringUtils.isEmpty(zkACLProviderClassName)) {
       try {
         log.info("Using ZkACLProvider: {}", zkACLProviderClassName);
-        return (ZkACLProvider) Class.forName(zkACLProviderClassName).getConstructor().newInstance();
+        ZkACLProvider zkACLProvider =
+            (ZkACLProvider) Class.forName(zkACLProviderClassName).getConstructor().newInstance();
+        zkACLProvider.setZkCredentialsInjector(zkCredentialsInjector);
+        return zkACLProvider;
       } catch (Throwable t) {
         // just ignore - go default
         log.warn(
             "VM param zkACLProvider does not point to a class implementing ZkACLProvider and with a non-arg constructor",
             t);
       }
     }
-    log.debug("Using default ZkACLProvider");
+    log.warn("Using default ZkACLProvider");
     return new DefaultZkACLProvider();
   }
 
+  public static final String ZK_CREDENTIALS_INJECTOR_CLASS_NAME_VM_PARAM_NAME =
```

Comment:
This is a weird place to declare such a constant.

Reply:
Followed the same existing logic as for _ZK_ACL_PROVIDER_CLASS_NAME_VM_PARAM_NAME_ and _ZK_CRED_PROVIDER_CLASS_NAME_VM_PARAM_NAME_.

Plus, I like keeping instance variables used only once close to the code where they are used.